ec3730/Data Feeds/DFService.swift

```diff
@@ -23,35 +23,42 @@ extension Service {
     var usage: Int {
         get {
             let search = NSPredicate(format: "serviceName like %@", argumentArray: [name])
+            // It's being used from the main thread no change is required
+            var result: [ServiceUsage]?
             if let context = AppDelegate.persistantStore?.viewContext {
-                let request = NSFetchRequest<ServiceUsage>(entityName: "ServiceUsage")
-                request.predicate = search
-
-                return (try? context.fetch(request).count) ?? 0
+                context.performAndWait {
```

Comment:
I'm thinking we should have this happen outside the getter/setter if there is a performance hit. Perhaps only fetch the first time and then if the data is "stale" (after a minute?)

What do you think?